### PR TITLE
Fixes #891: Caught more M2Crypto exceptions.

### LINKED ISSUE #891: Not all exceptions from SSL.Connection() are catched for M2Crypto
In `cim_http.py`, around line 577:
```
                # TODO 2/16 AM: Verify whether the additional exceptions
                #               in the M2Crypto code can
                #               really be omitted:
                #               Err.SSLError, SSL.SSLError, SSL.Checker.
                #               WrongHost,
                #               SSLTimeoutError
                except SSLError as arg:
                    raise ConnectionError(
                        "SSL error %s: %s" % (arg.__class__, arg))
```

These exceptions should also be catched to be on the safe side.

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see commit message.
Ready for review and merge.

### CODE REVIEW

File: docs/changes.rst [hunk 1]
```diff
@@ -161,6 +161,12 @@ Enhancements
 * Added `libxml2` operating system package as a dependency. It provides xmllint,
   which is used for testing.
 
+* Added catching of some exceptions M2Cryptro can raise that were not catched
```
Comment (KSchopmeyer) [inline docs/changes.rst]: changed catched to caught.  Don't you love english?

File: docs/changes.rst [hunk 2]
```diff
@@ -161,6 +161,12 @@ Enhancements
...
+  so far: SSL.SSLError, SSL.Checker.SSLVerificationError. These exceptions
+  are now transformed into `pywbem.ConnectionError` and will therefore be
+  catched by a caller of pywbem who is prepared for pywbem's own exceptions,
```
Comment (KSchopmeyer) [inline docs/changes.rst]: again caught

Comment (andy-maier) [reply docs/changes.rst]: DONE

File: docs/changes.rst (hunk 1 above)
Comment (andy-maier) [reply docs/changes.rst]: DONE

**Pull request merged**